test/cppsuite/test_harness/configuration_settings.h

```diff
@@ -39,22 +44,22 @@ class configuration {
         }
     }
 
-    std::string
+    const std::string &
     get_config()
     {
         return (_config);
     }
 
     /*
      * Wrapper functions for retrieving basic configuration values. Ideally the tests can avoid
-     * using the config item struct provided by wiredtiger. However if they still wish to use it the
-     * get and next functions can be used.
+     * using the config item struct provided by wiredtiger. However if they still wish to use it
+     * the get and next functions can be used.
      */
     int
```

Comment:
So we have decided to use error code instead of c++ return?

Reply:
May be it's hard to use st::optional due to compiler compatibility. Is it possible to just return std::string and use exception for handling?

Or this change is covered in another ticket?

Reply:
It will be handled later on, let's leave it as it is for now